doc/dev_manual/internals.rst

```diff
@@ -199,82 +199,30 @@ you expect the same functionality.
-            pass
-
-The ``save_as_file`` method is optional if you will never need to save
-a modified file from Ginga.
-To use your io handler with Ginga create your images like this:
-
-.. code-block:: python
-
-    from ginga.AstroImage import AstroImage
-    AstroImage.set_ioClass(MyIOHandler)
-    ...
-
-    image = AstroImage()
-    image.load_file(path)
-    ...
-    view.set_image(image)
-
-or you can override the io handler on a case-by-case basis:
-
-.. code-block:: python
-
-    from ginga.AstroImage import AstroImage
-    ...
-
-    image = AstroImage(ioclass=MyIOHandler)
-    image.load_file(path)
-    ...
-    view.set_image(image)
-
-You could also subclass AstroImage or BaseImage and implement your own
-I/O handling.
-
-.. note:: Both `naxispath` and `numhdu` are valid keyword arguments to
-          the load_file() method.
-
-          You probably want to treat `numhdu` as a kind of index into
-          your file, similarly to the meaning within a FITS file
-          (although you are free also to ignore it!).
-
-          If the user passes a valid numhdu (whatever that means to
-          your load_file method) you simply return that value that they
-          passed as the middle element of the return tuple. If they
-          passed None (default), then you return the index you used
-          to access the data area that you loaded.
-
-          You probably want to treat `naxispath` as any kind of path
-          that you would need to take to navigate through your kind of
-          data area selected by numhdu (above).  This is usually used to
-          describe the path through a data cube of N-dimensionality to
-          reach a 2D slice.
-
-          If the user passes a valid naxispath (whatever that means to
-          your load_file method) you simply return that value that they
-          passed. If they passed None (default), then you return
-          whatever path you used to access the data slice that you
-          returned.
+First of all, you can always create an ``AstroImage`` and assign its
+components for wcs and data.  However, assuming that you want to add a new
+type of loader into Ginga's file loading 
+
+Ginga's general file loading facility breaks the loading down into two
+phases: first, the file is identified by its ``magic`` signature
+(requires the Python module ``python-magic`` be installed), MIME
+type or filename extension.  Once the general category of file is known,
+methods in the specific I/O module devoted to that type are called to
+load the file data.
+
+The `ginga.util.loader` module is used to register file openers. An
+opener is a class that understand how to load data objects from a
+particular kind of file format.  You'll want to start by examining this
+module and especially looking at the examples at the bottom of that file
+for how openers are registered. 
+
+For implementing your own special opener, take a look at the
```

Comment:
I tried to clarify this in the docs, see what you think in the update I will push in a bit.  

I'm thinking we could do something with entry points (similar to what we have with the plugin template) so that you can install it separately, but for now, yes, you would have to do it via `ginga_config.py` if you want to add it to the reference viewer. If you are using the widget in your own code, of course it is easy to do it there.